communication/src/spark_protocol.cpp

```diff
@@ -1411,10 +1423,8 @@ bool SparkProtocol::handle_message(msg& message, token_t token, CoAPMessageType:
     case CoAPMessageType::DESCRIBE:
     {
         int desc_flags = DESCRIBE_ALL;
-        if (message.len > 8 && queue[8] <= DESCRIBE_ALL) {
+        if (message_padding_strip(queue, message.len) > 8 && queue[8] <= DESCRIBE_ALL) {
```

Comment:
I should have probably added a comment that this strips PKCS#7 padding. This is correctly handled in newer protocol implementation, see https://github.com/spark/firmware/blob/develop/communication/src/lightssl_message_channel.cpp#L76

It actually does fix an issue with https://github.com/spark/firmware/pull/1226 not working correctly as `message.len` is always `> 8` due to the padding.